### PR TITLE
Faster process for Marine Analysis Vrfy Task

### LINKED ISSUE #600: marine DA Verfy task takes too long in the ci
Comment out or make it faster.6 to 10mn is too long for a test.

### PR DESCRIPTION
**Pull request by apchoiCMD**
#### What this does:
- To improve a processing time on Marine DA VRFY Task
- Run a Python script `statePlotter(config)` concurrently

#### Profiling shows that the process of `vrfy` task is reduced to around ~240 to 250 sec

- Close #600

### CODE REVIEW

File: scripts/exgdas_global_marine_analysis_vrfy.py [hunk 1]
```diff
@@ -22,9 +22,11 @@
 import gen_eva_obs_yaml
 import marine_eva_post
 import diag_statistics
+from multiprocessing import Process
 from soca_vrfy import statePlotter, plotConfig
 import subprocess
 from datetime import datetime, timedelta
+import time
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: is this used?

File: scripts/exgdas_global_marine_analysis_vrfy.py [hunk 2]
```diff
@@ -41,164 +43,127 @@
...
+           plotConfig(grid_file=grid_file,
+                      data_file=os.path.join(com_ocean_history, f'{RUN}.t{gcyc}z.ocnf006.nc'),
+                      variables_horiz={'ave_ssh': [-1.8, 1.3],
+                                       'Temp': [-1.8, 34.0],
+                                       'Salt': [30, 38]},
+                      colormap='jet',
+                      comout=os.path.join(comout, 'vrfy', 'bkg'))]   # ocean surface background
+
+# Number of processes (adjust as needed)
+num_processes = len(configs)
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_vrfy.py:151]: Not sure what you mean by "adjust as needed".

File: scripts/exgdas_global_marine_analysis_vrfy.py [hunk 3]
```diff
@@ -41,164 +43,127 @@
...
+
+# Create a list to store the processes
+processes = []
+
+# Iterate over pairs of elements from configs and plot_functions
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: you're only iterating over the configs, so not pairs and no plot_functions

File: scripts/exgdas_global_marine_analysis_vrfy.py [hunk 4]
```diff
@@ -41,164 +43,127 @@
...
+for config in configs[:num_processes]:
+    process = Process(target=plot_marine_vrfy, args=(config,))  # Pass the entire config as an argument
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: I don't think this comment is useful, isn't `args=(config,)` self explanatory?

**Review by guillaumevernieres — COMMENTED** [review]
Thanks for looking into this @apchoiCMD . I'm sure there's a more `OO` way of solving this problem, but good enough for now! 
I have a few minor comments, but it also needs to be profiled properly before we merge.

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 1 above)
Comment (apchoiCMD) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: > is this used?

Removed, was used for profiling

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 2 above)
Comment (apchoiCMD) [reply scripts/exgdas_global_marine_analysis_vrfy.py:151]: Removed

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 4 above)
Comment (apchoiCMD) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: Removed

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 3 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: Looks like you forgot to commit your changes @apchoiCMD .

**Review by guillaumevernieres — APPROVED** [review]
Thanks @apchoiCMD , looks good. I assume you checked the figures and it all looks OK?

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 3 above)
Comment (apchoiCMD) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: > Looks like you forgot to commit your changes @apchoiCMD .

Yes, they look ok! Thanks for a good advice and reminder! @guillaumevernieres

**Pull request merged**